File: Dato abstracto/dato.java
import java.util.ArrayList;

class Persona {
    private String nombre;
    private String apellido1;
    private String apellido2;

    public Persona(String nombre, String apellido1, String apellido2) {
        this.nombre = nombre;
        this.apellido1 = apellido1;
        this.apellido2 = apellido2;
    }

    public String getNombreCompleto() {
        return nombre + " " + apellido1 + " " + apellido2;
    }
}

public class Main {
    public static void main(String[] args) {
        ArrayList<Persona> personas = new ArrayList<>();
        personas.add(new Persona("Emily", "sandoval", "Ramírez"));
        personas.add(new Persona("Carlos", "Ramírez", "Torres"));

        for (Persona persona : personas) {
            System.out.println(persona.getNombreCompleto());
        }
    }
}

File: Dato abstracto/dato.cpp
#include <iostream>
#include <vector>
#include <string>

class Persona {
private:
    std::string nombre;
    std::string apellido1;
    std::string apellido2;

public:
    Persona(std::string n, std::string a1, std::string a2)
        : nombre(n), apellido1(a1), apellido2(a2) {}

    std::string getNombreCompleto() const {
        return nombre + " " + apellido1 + " " + apellido2;
    }
};

int main() {
    std::vector<Persona> personas;
    personas.push_back(Persona("Emily", "sandoval", "Ramírez"));
    personas.push_back(Persona("Carlos", "Ramírez", "Torres"));

    for (const auto& persona : personas) {
        std::cout << persona.getNombreCompleto() << std::endl;
    }

    return 0;
}
